access-control-app/int_test/cmr/access_control/int_test/acl_crud_test.clj

```diff
@@ -1018,3 +1018,117 @@
+
+        acl2 (ac/create-acl (u/conn-context)
+                            (make-catalog-item "acl2" {:concept_ids [coll1 coll2 coll3]})
+                            {:token token})
+
+        acl3 (ac/create-acl (u/conn-context)
+                            (make-catalog-item "acl3" {:concept_ids [coll1 coll2 coll3]
+                                                       :entry_titles ["coll1 entry title"
+                                                                      "coll2 entry title"
+                                                                      "coll3 entry title"]})
+                            {:token token})
+
+        acl4 (ac/create-acl (u/conn-context)
+                            (make-catalog-item "acl4" {:concept_ids [coll1 coll2]
+                                                       :entry_titles ["coll2 entry title"
+                                                                      "coll3 entry title"]})
+                            {:token token})
+        expected-collection-identifier {:concept_ids [coll3 coll2 coll1]
+                                        :entry_titles ["coll3 entry title"
+                                                       "coll2 entry title"
+                                                       "coll1 entry title"]}
+        expected-collection-identifier2 {:concept_ids [coll4 coll3 coll2 coll1]
+                                         :entry_titles ["coll4 entry title"
+                                                        "coll3 entry title"
+                                                        "coll2 entry title"
+                                                        "coll1 entry title"]}]
+    (testing "create acls"
+      (are3 [concept-id]
+        (is (= expected-collection-identifier
+               (get-in (ac/get-acl (u/conn-context) concept-id {:token token})
+                       [:catalog_item_identity :collection_identifier])))
+
+        "Only entry-titles"
+        (:concept_id acl1)
+
+        "Only concept-ids"
+        (:concept_id acl2)
+
+        "Equal entry-titles and concept-ids"
+        (:concept_id acl3)
+
+        "Disjoint entry-titles and concept-ids"
+        (:concept_id acl4)))
+
+    (testing "update acls add collection"
+      (ac/update-acl (u/conn-context)
+                     (:concept_id acl4)
+                     (make-catalog-item "acl4" {:concept_ids [coll1 coll2 coll3 coll4]
+                                                :entry_titles ["coll1 entry title"
+                                                               "coll2 entry title"
+                                                               "coll3 entry title"]})
+
+                     {:token token})
+      (is (= expected-collection-identifier2
+             (get-in (ac/get-acl (u/conn-context) (:concept_id acl4) {:token token})
+                     [:catalog_item_identity :collection_identifier]))))
+
+    (testing "update acls remove collection"
```

Comment:
Because you are trying to sync entry-titles with concept-ids if you try and remove a collection from either via update it will always return when the collections are unioned.  This wouldn't affect the case where you are adding a collection to one or the other.